src/rinterface_extra.c

```diff
@@ -164,6 +164,31 @@ int igraphhcass2(int *n, int *ia, int *ib,
   return 0;
 } /* hcass2_ */
 
+
+SEXP R_igraph_hcass2(SEXP in, SEXP ia, SEXP ib) {
+  SEXP result;
+  int n = INTEGER(in)[0];
+  igraph_vector_int_t a, b;
+
+  if (igraph_vector_int_init(&a, n)) error("Out of memory");
```

Comment:
Perhaps this function is a great candidate for a first C++ wrapper -- correct memory handling will be much easier with RAII.